packages/amplify-codegen/src/commands/statements.js

```diff
@@ -55,15 +55,50 @@ async function generateStatements(context, forceDownloadSchema, maxDepth, withou
 
     try {
       fs.ensureDirSync(opsGenDirectory);
-      generate(schemaPath, opsGenDirectory, {
-        separateFiles: true,
-        language,
-        maxDepth: maxDepth || cfg.amplifyExtension.maxDepth,
+      const schemaData = loadSchema(schemaPath);
+      const generatedOps = generateGraphQLDocuments(schemaData, {
+        maxDepth: maxDepth || cfg.amplifyExtension.maxDepth
       });
+      await writeGeneratedStatements(language, generatedOps, opsGenDirectory, true);
       opsGenSpinner.succeed(constants.INFO_MESSAGE_OPS_GEN_SUCCESS + path.relative(path.resolve('.'), opsGenDirectory));
     } finally {
       opsGenSpinner.stop();
     }
   }
 }
+
+async function writeGeneratedStatements(language, generatedStatements, outputPath, separateFiles) {
+  const fileExtension = FILE_EXTENSION_MAP[language];
+  if(!generatedStatements) {
+    return;
+  }
+  if (separateFiles) {
+    ['queries', 'mutations', 'subscriptions', 'fragments'].forEach(op => {
+      const ops = generatedStatements[op];
+      if (ops.size) {
+        // This will change post top level codegen changes
+        fs.writeFileSync(path.resolve(path.join(outputPath, `${op}.${fileExtension}`)), JSON.stringify(Object.fromEntries(ops)));
```

Comment:
Is try/catch needed for a cleaner error message on CLI?

Reply:
This file is subject to change in upcoming PR. I've made some edits to get e2e passing but am going to re-visit this.